src/track/macro.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <QtCore>
+#include <memory>
+
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+#include "track/macroaction.h"
+#include "util/db/dbid.h"
+namespace proto = mixxx::track::io;
+
+constexpr int kMacrosPerChannel = 16;
+extern const QLoggingCategory kMacroLoggingCategory;
+
+/// A Macro stores a list of MacroActions as well as its current state and label.
+class Macro {
+  public:
+    static QByteArray serialize(const QList<MacroAction>& actions);
```

Comment:
I don't understand why these functions for serializing/deserializing one of the members are public? I would expect them to handle the entire class and not only an isolated aspect.

The minimum change is renaming them as serializeActions()/deserializeActions(), but there should be a more appropriate design.

Reply:
the actions _are_ the macro, this is what is needed to serialize it. Sure, they can be private, though I don't know what that adds.

Reply:
The macro entity is more than a list of actions. The list of actions is just the part that gets serialized.

Btw, repetition does not make wrong claims right. I am not going to explain everything twice.